src/Gui/NotificationArea.cpp

```diff
@@ -0,0 +1,991 @@
+{
+    hGrp = App::GetApplication().GetParameterGroupByPath(
+        "User parameter:BaseApp/Preferences/NotificationArea");
+
+    parameterMap = {
+        {"NotificationAreaEnabled",
+         [this](const std::string& string) {
+             auto enabled = hGrp->GetBool(string.c_str(), true);
+             if (!enabled)
+                 notificationArea->deleteLater();
+         }},
+        {"NonIntrusiveNotificationsEnabled",
+         [this](const std::string& string) {
+             auto enabled = hGrp->GetBool(string.c_str(), true);
+             notificationArea->pImp->notificationsDisabled = !enabled;
+         }},
+        {"NotificationTime",
+         [this](const std::string& string) {
+             auto time = hGrp->GetInt(string.c_str(), 20) * 1000;
+             if (time < 0)
+                 time = 0;
+             notificationArea->pImp->notificationExpirationTime = static_cast<unsigned int>(time);
+         }},
+        {"MinimumOnScreenTime",
+         [this](const std::string& string) {
+             auto time = hGrp->GetInt(string.c_str(), 5) * 1000;
+             if (time < 0)
+                 time = 0;
+             notificationArea->pImp->minimumOnScreenTime = static_cast<unsigned int>(time);
+         }},
+        {"MaxOpenNotifications",
+         [this](const std::string& string) {
+             auto limit = hGrp->GetInt(string.c_str(), 15);
+             if (limit < 0)
+                 limit = 0;
+             notificationArea->pImp->maxOpenNotifications = static_cast<unsigned int>(limit);
+         }},
+        {"MaxWidgetMessages",
+         [this](const std::string& string) {
+             auto limit = hGrp->GetInt(string.c_str(), 1000);
+             if (limit < 0)
+                 limit = 0;
+             notificationArea->pImp->maxWidgetMessages = static_cast<unsigned int>(limit);
+         }},
+        {"AutoRemoveUserNotifications",
+         [this](const std::string& string) {
+             auto enabled = hGrp->GetBool(string.c_str(), true);
+             notificationArea->pImp->autoRemoveUserNotifications = enabled;
+         }},
+    };
+
+    for (auto& val : parameterMap) {
+        auto string = val.first;
+        auto update = val.second;
+
+        update(string);
+    }
+
+    hGrp->Attach(this);
+}
+
+NotificationArea::ParameterObserver::~ParameterObserver()
+{
+    hGrp->Detach(this);
+}
+
+void NotificationArea::ParameterObserver::OnChange(Base::Subject<const char*>& rCaller,
+                                                   const char* sReason)
+{
+    (void)rCaller;
+
+    auto key = parameterMap.find(sReason);
+
+    if (key != parameterMap.end()) {
+        auto string = key->first;
+        auto update = key->second;
+
+        update(string);
+    }
+}
+
+/************************* Notification Area *****************************************/
+
+NotificationArea::NotificationArea(QWidget* parent)
+    : QPushButton(parent)
+{
+    // QPushButton appearance
+    setText(0);
```

Comment:
This raises an error that `QString(const char*)` is private. Should be `setText(QString())`